File: app/src/main/java/com/vaibhav/moviedetails/application/MovieDetailsApplication.java
package com.vaibhav.moviedetails.application;

import android.content.Context;
import android.support.multidex.MultiDex;
import android.support.multidex.MultiDexApplication;

import com.vaibhav.moviedetails.commons.LogUtils;
import com.vaibhav.moviedetails.commons.ResourceUtils;

/**
 * @author Vaibhav Bhandula on 09/10/18.
 */
public class MovieDetailsApplication extends MultiDexApplication {

    @Override protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        try {
            MultiDex.install(this);
        } catch (Exception e) {
            LogUtils.printStackTrace(e);
        }
    }

    @Override public void onCreate() {
        super.onCreate();
        initialize();
    }

    private void initialize() {
        ResourceUtils.initialize(getApplicationContext());
    }
}

File: app/src/main/java/com/vaibhav/moviedetails/application/MovieDetailsApplication.kt
package com.vaibhav.moviedetails.application

import android.arch.persistence.room.Room
import android.content.Context
import android.support.multidex.MultiDex
import android.support.multidex.MultiDexApplication
import com.vaibhav.moviedetails.commons.LogUtils
import com.vaibhav.moviedetails.db.MovieDatabase

/**
 * @author Vaibhav Bhandula on 08/04/19.
 */
class MovieDetailsApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        appDatabase = Room.databaseBuilder(applicationContext,
                MovieDatabase::class.java, "movieDatabase.db").build()
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        try {
            MultiDex.install(this)
        } catch (e: Exception) {
            LogUtils.printStackTrace(e)
        }
    }

    companion object {
        private lateinit var appDatabase: MovieDatabase

        fun getMovieDao() = appDatabase.movieDao()
    }

}
